### PR TITLE
util: Alpha Python timeline utility updates

### PR DESCRIPTION
**Pull request by JLGarber**
It's all bad, but it seemed to work on the alpha logs I was given. I'm not particularly attached to anything I've implemented here, this was mostly an attempt to get *something* in place for later improvement.

For consistency with converted old logs, I figured it made most sense to format `make_timeline` output the same way, as `1[56]`.

### CODE REVIEW

File: util/encounter_tools.py [hunk 1]
```diff
@@ -256,23 +256,34 @@ def is_tl_line_syncmatch(line):
...
 def is_tl_line_log(poss_match):
-    return re.search(r"^\s*00:([0-9]*):(.*$)", poss_match)
+    return re.search(
+        r"\s00:(?P<id>[0-9]*):(?P<entity>[^:]*):(?P<message>[^\/]*)\s?\/\s", poss_match
+    )
 
 
 def is_tl_line_adds(poss_match):
-    return re.search(r"Added new combatant (.*$)", poss_match)
+    return re.search(r"\s*03:........:(?P<entity>[^:]*):", poss_match)
```
Comment (quisquous) [inline util/encounter_tools.py:280]: Why do some of these lines have `\s*` and others just `\s`? I think it should just be a single space in all cases, right?

File: util/make_timeline.py [hunk 2]
```diff
@@ -371,7 +371,7 @@ def main(args):
         if entry["line_type"] == "34":
             output_entry = '{position:.1f} "{targetable}"'.format(**entry)
```
Comment (quisquous) [inline util/make_timeline.py:372]: Are these targetable lines still ok?

File: util/test_timeline.py [hunk 3]
```diff
@@ -39,38 +39,69 @@ def load_timeline(timeline):
...
             entry["special_type"] = False
 
             begincast_match = e_tools.is_tl_line_begincast(sync_match.group(1))
             if begincast_match:
                 entry["special_type"] = "begincast"
                 entry["special_line"] = "20"
-                entry["cast_id"] = begincast_match.group(1)
-                entry["caster_name"] = begincast_match.group(2)
+                entry["cast_id"] = begincast_match.group("id")
+                entry["caster_name"] = begincast_match.group("source")
```
Comment (quisquous) [inline util/test_timeline.py:57]: Did you verify that all of these log types: cast, begincast, buff, log, adds, headmarker all match properly? Mostly just looking for "have each of these conditional blocks been run at least once and seem to work" level of testing.

File: util/make_timeline.py (hunk 2 above)
Comment (JLGarber) [reply util/make_timeline.py:372]: These don't include syncs ever, it just writes it in as `1000.0 "--targetable--"`. (When I get to the Typescript re-write, I may add the ability to sync these optionally, but it's not worth trying to add that to the Python version here.)

File: util/encounter_tools.py (hunk 1 above)
Comment (JLGarber) [reply util/encounter_tools.py:284]: It should indeed.

File: util/test_timeline.py [hunk 4]
```diff
@@ -236,11 +262,7 @@ def test_match(event, entry):
         elif entry["special_type"] == "battlelog" and event.startswith(entry["special_line"]):
             # Matching this format generically:
             # 00|2019-01-12T18:08:14.0000000-05:00|0839||The Realm of the Machinists will be sealed off in 15 seconds!|
-            log_match = re.search(
-                "^00\|[^\|]*\|{}\|[^\|]*\|{}".format(entry["logid"], entry["line"]),
-                event,
-                re.IGNORECASE,
-            )
+            log_match = re.search(entry["regex"], event, re.IGNORECASE,)
```
Comment (JLGarber) [inline util/test_timeline.py:265]: *This* little annoyance right here was what made `gainsEffect` syncs perpetually fail. We can't rely on the casing of the `effectId` in the timeline file to match that in the network log. ~~It took me "only" half an hour of "WHY WON'T THIS WORK? IT MATCHES!" to debug~~

File: util/test_timeline.py (hunk 3 above)
Comment (JLGarber) [reply util/test_timeline.py:57]: Verified now. SO much nonsense.

File: util/test_timeline.py [hunk 5]
```diff
@@ -224,9 +252,7 @@ def test_match(event, entry):
         elif entry["special_type"] == "applydebuff" and event.startswith(entry["special_line"]):
             # Matching this format generically:
             # |Dadaluma Simulation|0.00|E0000000||4000AE96|Guardian
-            buff_match = re.search(
-                "\|{}\|([^\|]*\|){{4}}{}".format(entry["buff_name"], entry["buff_target"]), event
-            )
+            buff_match = re.search(entry["regex"], event, re.IGNORECASE)
```
Comment (JLGarber) [inline util/test_timeline.py:255]: *This* little annoyance right here was what made `gainsEffect` syncs perpetually fail. We can't rely on the casing of the `effectId` in the timeline file to match that in the network log. ~~It took me "only" half an hour of "WHY WON'T THIS WORK? IT MATCHES!" to debug~~

Comment (quisquous) [reply util/test_timeline.py:255]: Oh nooo.

If it makes you feel any better, everything is uppercase in the alpha plugin (but we probably need to keep this indefinitely to support old network log files).

**Pull request merged**